### PR TITLE
style(example-usage): add linting and fix linting errors

### PR DESCRIPTION
**Pull request by JaeAeich**
This commit extends eslint-config-elixir to example-usage and fixes linting errors in that directory, moreover now npm run lint can be run without errors.

## Description

This commit adds an example ESLint configuration and resolves linting errors in the 'example-usage' app. Previously, running 'npm run lint' resulted in errors, but now the code has been properly linted. Additionally, the global linting process no longer reports any errors.

## Checklist

> Please go through the following checklist to ensure that your change is ready
> for review. Please do not forget to double check the list after you have
> modified your PR, e.g., if you have added commits to address reviewer
> comments or to fix failing automated checks. Please check items also if they
> do not apply to your change, e.g., if your change does not require an update
> of the user-facing documentation, then still check the box. Generally, PRs
> are only reviewed when all boxes are ticked off and all automated checks pass
> (use the comment section below if you believe that your PR is ready to be
> merged even though not all boxes were ticked off).

- [x] My code follows the [contributing guidelines][contributing] of this
      project, including, in particular, with regard to any style guidelines
- [x] The title of my PR complies with the [Conventional Commits
      specification][conventional-commits]; in particular, it clearly indicates
      that a change is a breaking change
- [x] I am aware that all my commits will be squashed into a single commit,
      using the PR title as the commit message.
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have updated the user-facing documentation to describe any new or
      changed behavior
- [ ] I have added type annotations for all function/class/method interfaces
      or updated existing ones
- [ ] I have provided [JSDoc][jsdoc] block tags for all
      functions/classes/methods or updated existing ones
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature
      works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage
- [x] I have asked the [@all-contributors bot][all-contributors-bot] to
      acknowledge my contributions by commenting on this PR with a request of
      the form `@all-contributors please add @YOUR_GH_HANDLE for TYPE_1,
    TYPE_2, ...`, where `TYPE_1` etc. refer to [contribution types supported
      by the All Contributors Specification][all-contributors-types] OR I do
      not want my contributions to be acknowledged

## Comments
Unchecked boxes above are not required for this PR.

[all-contributors-bot]: https://allcontributors.org/docs/en/bot/overview
[all-contributors-types]: https://allcontributors.org/docs/en/emoji-key
[contributing]: CONTRIBUTING.md
[conventional-commits]: https://www.conventionalcommits.org/en/v1.0.0/
[jsdoc]: https://jsdoc.app/index.html

### CODE REVIEW

**Comment by JaeAeich** [comment]
@git-anurag-hub I see a redundant file in eslint-config-elixir, namely `.eslintrc.js` and an unused plugin `eslint-plugin-import` do they serve a purpose or should I remove it, maybe adding the plugin would make more sense but since it would be applied to the base configuration I would like to know your opinion.

File: apps/example-usage/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  extends: ['elixir'],
```
Comment (git-anurag-hub) [inline apps/example-usage/.eslintrc.js:2]: It should also extend `next` config!

File: apps/example-usage/package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
     "@types/react": "18.2.5",
     "@types/react-dom": "18.2.3",
     "elixir-ui": "*",
+    "eslint-config-elixir":"*",
```
Comment (git-anurag-hub) [inline apps/example-usage/package.json:17]: Install `eslint` too as dev deps!

File: apps/example-usage/src/app/layout.tsx [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-  title: "Create Next App",
-  description: "Generated by create next app",
+  title: 'Create Next App',
+  description: 'Generated by create next app',
 };
 
 export default function RootLayout({
   children,
 }: {
+  // eslint-disable-next-line no-undef
```
Comment (git-anurag-hub) [inline apps/example-usage/src/app/layout.tsx:14]: Why this?

**Review by git-anurag-hub — CHANGES_REQUESTED** [review]

**Comment by git-anurag-hub** [comment]
`.eslintrc.js` is not redundant as it lints the `eslint-config-elixir` package. Ohh missed the `eslint-plugin-import` don't remove that I will do that in the next global fixes update.

**Pull request merged**